lib/puppet/provider/service/smf.rb

```diff
@@ -71,6 +72,8 @@ def startcmd
     case self.status
     when :maintenance
```

Comment:
I think maintenance and degraded could be combined into the same when, as they have the same action.

```
when :maintenance, :degraded
  [command(:adm), :clear, @resource[:name]]
```